src/odemis/driver/test/actuator_test.py

```diff
@@ -2013,5 +2013,43 @@ def test_calculation(self):
         self.assertEqual(self.dev.rotation.value, -math.pi / 4)
 
 
+class TestDualChannelPositionSensorWithStage(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Construct DualChannelPositionSensor object and its dependencies.
+        """
+        cls.sensor = smaract.Picoscale(name="Stage Metrology",
+                                   role="metrology",
+                                   ref_on_init=False,
+                                   locator="fake",
+                                   channels={'x1': 0, 'x2': 1, 'y1': 2},
+                                   )
+        cls.stage = tmcm.TMCLController(name="test", role="test",
+                                    port="/dev/fake3",
+                                    axes=["x", "y"],
+                                    ustepsize=[5.9e-9, 5.8e-9],
+                                    rng=[[-1e-3, 1e-3], [0, 1e-3]],
+                                    refproc="Standard")
+        cls.dev = DualChannelPositionSensor(name="TestDualChannelPositionSensor",
+                                            role="stage-position",
+                                            dependencies={"sensor": cls.sensor,
+                                                          "stage": cls.stage},
+                                            channels={'x': ['x1', 'x2'], 'y': 'y1'},
+                                            distance=1e-6,
+                                            ref_on_init="if necessary"
```

Comment:
Maybe you need to add an additional test for other options (specially "always").